### PR TITLE
Fix regression introduced by 08e4b076be6402a4f98a2e082b1a292756c570ff.

### PR DESCRIPTION
**Pull request by normanmaurer**

### CODE REVIEW

**Comment by normanmaurer** [comment]
@netkins build

File: transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollEventLoop.java [hunk 1]
```diff
@@ -313,15 +313,15 @@ private void processReady(EpollEventArray events, int ready) {
                 if (ch != null) {
                     AbstractEpollUnsafe unsafe = (AbstractEpollUnsafe) ch.unsafe();
 
-                    // First check if EPOLLIN was set, in this case we do not need to check for
-                    // EPOLLRDHUP as EPOLLIN will handle connection-reset case as well.
-                    if ((ev & Native.EPOLLIN) != 0) {
+                    // First check if EPOLLRDHUP was set, this will notify us for connection-reset in which case
+                    // we may close the channel.
+                    if ((ev & Native.EPOLLRDHUP) != 0) {
```
Comment (Scottmitch) [inline transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollEventLoop.java]: Can you explain the change in order in these conditionals? Is it possible the channel may be closed before we actually do the read in the next conditional?

Comment (normanmaurer) [inline transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollEventLoop.java]: will do once I know it is the cause :) Just opened the PR so the CI picks it up.

**Comment by normanmaurer** [comment]
Hopefully the comments clarify it.

**Comment by Scottmitch** [comment]
@normanmaurer - Not really. The purpose of https://github.com/netty/netty/commit/08e4b076be6402a4f98a2e082b1a292756c570ff was to `read from socket when EPOLLRDUP is received` and thus changed the read handling before the `EPOLLRDUP`.  Are you saying this no longer makes sense?

**Comment by normanmaurer** [comment]
@Scottmitch my assumption of not need EPOLLOUT when EPOLLIN was processed was wrong. I changed the order as when EPOLLOUT is done before chances are good that EPOLLIN is not needed anymore as the Channel is closed.

**Pull request closed without merging**